src/lib/comms/sol-lwm2m-client.c

```diff
@@ -1362,10 +1369,17 @@ handle_resource(void *data, struct sol_coap_server *server,
         goto exit;
     }
 
-    if (client->supports_access_control) {
-        r = get_server_id_by_link_addr(&client->connections, cliaddr, &server_id);
-        header_code = SOL_COAP_RESPONSE_CODE_INTERNAL_ERROR;
-        SOL_INT_CHECK_GOTO(r, < 0, exit);
+    r = get_server_id_by_link_addr(&client->connections, cliaddr, &server_id);
+    header_code = SOL_COAP_RESPONSE_CODE_INTERNAL_ERROR;
+    SOL_INT_CHECK_GOTO(r, < 0, exit);
+
+    if (path_size >= 1 &&
```

Comment:
I would still need the first check (`path_size >= 1`) to enforce `obj_ctx != NULL`, but I guess your version is shorter/more clear.
Changed to:
```C
    if (path_size >= 1 && !client->is_bootstrapping &&
        obj_ctx->obj->id == SECURITY_OBJECT_ID) {
        SOL_WRN("Only the Bootstrap Server is allowed to access the Security Object."
            " Server ID %" PRId64 " trying to access it", server_id);
        header_code = SOL_COAP_RESPONSE_CODE_UNAUTHORIZED;
        goto exit;
    }
```